src/routers/sessions.py

```
@@ -460,12 +462,24 @@
+        ..., description="Message search parameters "
+    ),
     db: AsyncSession = db,
 ):
+    query, use_semantic_search = search.query, search.use_semantic_search
+    embed_messages_enabled = settings.LLM.EMBED_MESSAGES
+    if use_semantic_search and not embed_messages_enabled:
+        raise DisabledException(
+            "Semantic search requires EMBED_MESSAGES flag to be enabled"
+        )
+
     """Search a Session"""
```

Pylint (3.3.7)

[warning] String statement has no effect

(W0105)